sdk/ai/azure-ai-inference/tests/test_model_inference_client.py

```diff
@@ -795,3 +810,1090 @@ def test_embeddings_on_chat_completion_endpoint(self, **kwargs):
+            assert AIInferenceApiInstrumentor().is_instrumented() == False
+        except RuntimeError as e:
+            exception_caught = True
+            print(e)
+        client.close()
+        assert exception_caught == False
+
+    @ServicePreparerChatCompletions()
+    def test_instrumenting_twice_causes_exception(self, **kwargs):
+        client = self._create_chat_client(**kwargs)
+        exception_caught = False
+        instrumented_once = False
+        try:
+            AIInferenceApiInstrumentor().instrument()
+            instrumented_once = True
+            AIInferenceApiInstrumentor().instrument()
+        except RuntimeError as e:
+            exception_caught = True
+            print(e)
+        client.close()
+        assert instrumented_once == True
+        assert exception_caught == True
+
+    @ServicePreparerChatCompletions()
+    def test_uninstrumenting_uninstrumented_causes_exception(self, **kwargs):
+        client = self._create_chat_client(**kwargs)
+        exception_caught = False
+        try:
+            AIInferenceApiInstrumentor().uninstrument()
+        except RuntimeError as e:
+            exception_caught = True
+            print(e)
+        client.close()
+        assert exception_caught == True
+
+    @ServicePreparerChatCompletions()
+    def test_uninstrumenting_twise_causes_exception(self, **kwargs):
+        client = self._create_chat_client(**kwargs)
+        exception_caught = False
+        uninstrumented_once = False
+        try:
+            AIInferenceApiInstrumentor().instrument()
+            AIInferenceApiInstrumentor().uninstrument()
+            uninstrumented_once = True
+            AIInferenceApiInstrumentor().uninstrument()
+        except RuntimeError as e:
+            exception_caught = True
+            print(e)
+        client.close()
+        assert uninstrumented_once == True
+        assert exception_caught == True
+
+    @ServicePreparerChatCompletions()
+    def test_chat_completion_tracing_content_recording_disabled(self, **kwargs):
+        self.modify_env_var(CONTENT_TRACING_ENV_VARIABLE, "False")
+        client = self._create_chat_client(**kwargs)
+        processor, exporter = self.setup_memory_trace_exporter()
+        AIInferenceApiInstrumentor().instrument()
+        response = client.complete(
+            messages=[
+                sdk.models.SystemMessage(content="You are a helpful assistant."),
+                sdk.models.UserMessage(content="What is the capital of France?"),
+            ],
+        )
+        processor.force_flush()
+        spans = exporter.get_spans_by_name_starts_with("chat ")
+        if len(spans) == 0:
+            spans = exporter.get_spans_by_name("chat")
+        assert len(spans) == 1
+        span = spans[0]
+        expected_attributes = [('gen_ai.operation.name', 'chat'),
+                               ('gen_ai.system', 'az.ai.inference'),
+                               ('gen_ai.request.model', ''),
```

Comment:
let's validate that attribute values are populated

Reply:
added more validation